File: fixtures/python/process-exec-dynamic/negative.py
"""Builds argv for somebody else to run, and runs nothing itself."""

import shlex

TEMPLATE = "rm -rf {path}"


def build(path):
    # Must NOT fire: quoting a command is not spawning one.
    return shlex.quote(TEMPLATE.format(path=path))


def plan(steps):
    # Must NOT fire: a list of command words is data until something spawns it.
    return [shlex.split(step) for step in steps]

File: fixtures/javascript/process-exec-dynamic/negative.js
// Builds argv for somebody else to run, and runs nothing itself.

function build(pkg) {
  // Must NOT fire: assembling a command string is not spawning one.
  return `npm install ${pkg}`;
}

function plan(steps) {
  // Must NOT fire: command words are data until something spawns them.
  return steps.map((step) => step.split(" "));
}

module.exports = { build, plan };
